CTP-4673 Fix block displays future timed forum posts

leonstr opened the pull request.

leonstr committed:
CI fixes

leonstr committed:
PostgreSQL compatibility fix

aydevworks commented on block_forumfeed.php:
```diff
@@ -100,6 +100,7 @@ public function forum_posts(): string {
             }, $courses);
 
             list($incourses, $params) = $DB->get_in_or_equal($courseids, SQL_PARAMS_NAMED);
+            $now = time();
```
Probably worth using this https://moodledev.io/docs/4.5/apis/core/clock instead of time();

leonstr replied:
Thanks @aydevworks, I agree. I've reworked the commit to use this (used forced push so apologies if you have to re-clone).

leonstr committed:
CTP-4673 Respect discussion display period

Forum discussions may have a "Display start"/"Display end" setting.
Previously this block ignored these settings potentially showing posts
the user should not be able to see and did not have access to.  We now
exclude such discussions from the block.
